packages/next-app/src/hooks/useTransaction.ts

```diff
@@ -260,8 +274,39 @@ export function useInternalBalancesTransaction({
 
   const { data, write } = useVaultManageUserBalance(config);
 
+  //function to prepare depoist transaction
+  async function approveToken() {
+    if (!submitData) return;
+    setTransactionStatus(TransactionStatus.AUTHORIZING);
+    setNotification(
+      NOTIFICATION_MAP_INTERNAL_BALANCES[TransactionStatus.AUTHORIZING]
+    );
+    const config = await prepareWriteContract({
+      address: submitData?.tokenAddress as `0x${string}`,
+      abi: erc20ABI,
+      functionName: "approve",
+      args: [
+        vaultAddress[5],
+        parseFixed(
+          submitData?.tokenAmount ? submitData.tokenAmount : "0",
+          tokenDecimals
+        ),
+      ],
+    });
```

Comment:
👁️ could we generate a wagmi hook for this? We'd get warned by the usepreparedtransactionwrite if it'd fail